element-hq/element-x-android
# Add room badges

## Type of change

- [x] Other: UI improvements

## Content

- Add generic `Badge` component based on the [Compound designs](https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=1960-476&mode=design&t=IH4MlCssOB8sIoQr-4).
- Add `RoomBadge` with the values needed for [the room details screen](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=1787-52822&mode=design&t=ibEN84jYOxZOeguX-4) (Encrypted, Not encrypted, Public room).

## Motivation and context

Update the designs so they match the current ones in Figma.

## Screenshots / GIFs

|Before|After|
|-|-|
|![](https://media.githubusercontent.com/media/element-hq/element-x-android/4c1bd2d45bb94a4531a96c5b9e2bd1d35b79373e/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.roomdetails.impl_RoomDetails_null_RoomDetails--0_2_null_9%2CNEXUS_5%2C1.0%2Cen%5D.png)|![](https://media.githubusercontent.com/media/element-hq/element-x-android/54485b694065beb27fd533282d8ec749bf160afb/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.roomdetails.impl_RoomDetails_null_RoomDetails--0_2_null_9%2CNEXUS_5%2C1.0%2Cen%5D.png)|

## Tests

- Open several rooms.
- Check the new badges are displayed when necessary.

## Tested devices

- OS version(s): 14

## Checklist
